pkg/util/listblocks/listblocks.go

```diff
@@ -20,25 +21,43 @@ import (
 	"github.com/grafana/mimir/pkg/storage/tsdb/block"
 )
 
-// LoadMetaFilesAndDeletionMarkers reads the bucket and loads the meta files for the provided user.
+type MarkerDetails struct {
```

Comment:
Here in order to reuse fetchMarkerDetails function, create a type could be return for both delete marker and no-compact marker.